backend/wmg/api/rollup.py

```diff
@@ -0,0 +1,130 @@
+import owlready2
+import pandas as pd
+import numpy as np
+import numba as nb
+from functools import lru_cache
+from backend.wmg.data.constants import CL_BASIC_PERMANENT_URL_OWL
+
+
+# ontology object
+onto = None
+
+
+def find_descendants_per_cell_type(cell_types):
+    """
+    Find descendants for each cell type in the input list.
+
+    Parameters
+    ----------
+    cell_types : list
+        List of cell types (cell type ontology term IDs) to find descendants for
+
+    Returns
+    -------
+    descendants_per_cell_type : list
+        List of lists of descendants for each cell type in the input list.
+    """
+
+    global onto
+    if not onto:
+        onto = owlready2.get_ontology(CL_BASIC_PERMANENT_URL_OWL)
+        onto.load()
+
+    # cache finding descendants per cell type
+    @lru_cache(maxsize=None)
```

Comment:
Any reason for `maxsize=None`? Since this runs on a webserver, caches should always be bounded.

Reply:
That said, I understand that the number of values is limited and it's probably not an issue.

Reply:
Yeah, I could technically precompute all the descendants for all the cell types and the resulting object wouldn't be that large.